File: 100 Same Tree/C/solution.c
#include <stdlib.h>

#include "solution.h"

bool isSameTree(struct TreeNode* p, struct TreeNode* q) {
    if (p == NULL && q == NULL) {
        return true;
    }
    else if (p == NULL || q == NULL) {
        return false;
    }

    bool left = isSameTree(p->left, q->left);
    bool right = isSameTree(p->right, q->right);
    return left && right && p->val == q->val;
}

File: 100 Same Tree/Python/Solution.py
from typing import Optional

from TreeNode import TreeNode


class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p and not q:
            return True
        elif not p or not q:
            return False
        
        left: bool = self.isSameTree(p.left, q.left)
        right: bool = self.isSameTree(p.right, q.right)
        return left and right and p.val == q.val